cub/cub/util_type.cuh

```diff
@@ -339,22 +339,23 @@ struct AlignBytes
   };
 
   /// The "truly aligned" type
-  typedef T Type;
+  using Type = T;
 };
 
 // Specializations where host C++ compilers (e.g., 32-bit Windows) may disagree
 // with device C++ compilers (EDG) on types passed as template parameters through
 // kernel functions
 
-#  define __CUB_ALIGN_BYTES(t, b)  \
-    template <>                    \
-    struct AlignBytes<t>           \
-    {                              \
-      enum                         \
-      {                            \
-        ALIGN_BYTES = b            \
-      };                           \
-      typedef __align__(b) t Type; \
+#  define __CUB_ALIGN_BYTES(t, b)                                                                  \
+    template <>                                                                                    \
+    struct AlignBytes<t>                                                                           \
+    {                                                                                              \
+      enum                                                                                         \
+      {                                                                                            \
+        ALIGN_BYTES = b                                                                            \
+      };                                                                                           \
+      typedef __align__(b) t Type;                                                                 \
+      /* TODO(bgruber): rewriting the above to using Type __align__(b) = t; does not compile :S */ \
```

Comment:
This is a `typedef` I couldn't rewrite. GCC/clang would accept an `alignas` for a `using` declaration, but I could not find a combination that works for nvcc.